pylib/gyp/msvs_emulation.py

```diff
@@ -9,20 +9,22 @@
 
 import collections
 import os
+import pickle
 import re
 import subprocess
 import sys
+import time
+import hashlib
+import traceback
 
 from gyp.common import OrderedSet
 import gyp.MSVSUtil
 import gyp.MSVSVersion
-
-try:
-  # basestring was removed in python3.
-  basestring
-except NameError:
+from gyp import DebugOutput, DEBUG_GENERAL
+if 'basestring' not in __builtins__:
```

Comment:
This will raise a Flake and pylint _undefined name_ that the __try / except NameError__ approach will not.

Reply:
IIUC since it's a key lookup it shouldn't (also seems more "polite" then the actual name lookup in the try/except idiom).
Linters seem to agree.